pkg/applicationserver/observability.go

```diff
@@ -282,8 +282,9 @@ func registerDropUp(ctx context.Context, msg *ttnpb.ApplicationUp, err error) {
 func registerUplinkLatency(ctx context.Context, msg *ttnpb.ApplicationUplink) {
 	asMetrics.nsAsUplinkLatency.WithLabelValues(ctx).Observe(time.Since(*msg.ReceivedAt).Seconds())
 	for _, meta := range msg.RxMetadata {
-		if meta.Time != nil && !meta.Time.IsZero() {
-			asMetrics.gtwAsUplinkLatency.WithLabelValues(ctx).Observe(time.Since(*meta.Time).Seconds())
+		stdTime := ttnpb.StdTime(meta.Time)
+		if meta.Time != nil && !stdTime.IsZero() {
```

Comment:
You can do `if stdTime := ttnpb.StdTime(meta.Time) ; stdTime != nil {`. (and drop the assignment above).